### DISCUSSION TITLE
Axios is not defined

### DISCUSSION THREAD

**Discussion by LuisCastilloV98** [directus/directus#6352] [Archive]
1) The issue and what you expected to happen
    I have been trying to reproduce your examples of asynchronous calls but I have not succeeded, which I intend to make call the api that has the items but I do not succeed because it appears to me as undefined axis, the code in the hook is:

```// extensions/hooks/sync-with-external/index.js

module.exports = function registerHook({ services, exceptions }) {
    
	const { MailService } = services;
	const { ServiceUnavailableException, ForbiddenException } = exceptions;

	return {
		// Sync with external recipes service, cancel creation on failure
		'items.create.before': async function (input, { collection, schema }) {
			// if (collection !== 'terminals') return input;

            console.log("*****************Flag*********************");
            console.log(input);
            console.log(collection);
        
			await checkToken("123")
			input.token = randomStr(12, getValidStrings());

			return input;
        },


        // 'cron(*/15 * * * * *)': async function () {
		// 	console.log("Holi");
		// },
	};
};


async function checkToken(newToken){

    var result = await axios.get('http://localhost:8055/items/terminals');
    console.log(result)
}

function randomStr(len, arr) {
    var ans = '';
    for (var i = len; i > 0; i--) {
        var randomPos = Math.floor(Math.random() * arr.length)
        ans += arr[randomPos];
    }
    console.log(ans)
    return ans;
}

function getUpperLetters(){
    const caps = [...Array(26)].map((val, i) => String.fromCharCode(i + 65));
    return caps
    // return caps.concat(caps.map(letter => letter.toLowerCase()));
}

function getNumbers(){
    var result = []
    for (let index = 0; index < 10; index++) {
        result.push(index.toString())
        
    }
    return result
}

function getValidStrings(){
    var result = getUpperLetters();
    result = result.concat(getNumbers())
    return result
}
```

2) Exact steps to reproduce this issue
When the event that calls the previous code occurs, the following error occurs:
11:27:51 ≡ƒÜ¿ axios is not defined
ReferenceError: axios is not defined

3) Your environment:
Directus: Lasttversion 9
DBMS: MariaDB
Deployment are: CLI.
Browser are: Google Chrome.

**Comment by juancarlosjr97** [comment]
@LuisCastilloV98 whatever you are using inside your hooks you have to import it or use the context. If you want to use `axios`, you need to do the following inside your hook.

```
module.exports = () => {
  return {
    init: async () => {
      const axios = require("axios");
      const response = await axios.get("endpoint...");
      ....
    },
  };
};
```

`axios` is part of Directus so it can be imported from the extensions.

**Reply by LuisCastilloV98** [reply]
Thanks, it works perfectly, it would be cool if you added this in your documentation, regards.

**Reply by paescuj** [reply]
Already done (#6360) 😃
Can you mark the discussion as answered?

**Comment by paescuj** [comment]
Should we include this in the example under https://docs.directus.io/guides/api-hooks/?

**Reply by juancarlosjr97** [reply]
Yes, feel free to create a PR to update the docs @paescuj. There are a few examples using `axios` though

**Reply by rijkvanzanten** [reply]
Yeah if the example includes usage of `axios`, we should add the require in the same example as well 👍🏻

**Reply by paescuj** [reply]
I meant the `require` part. Will do 😃
